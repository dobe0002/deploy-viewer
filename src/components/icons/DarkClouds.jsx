// From: https://thenounproject.com/adamwhitcroft/uploads/?i=2600

import React from 'react';
import PropTypes from 'prop-types';

const DarkClouds = ({ size, color, alt }) => {
  return (
    <svg
      height={size}
      width={size}
      fill={color}
      aria-label={alt}
      xmlns='http://www.w3.org/2000/svg'
      x='0px'
      y='0px'
      viewBox='0 0 100 68.18'
      enableBackground='new 0 0 100 68.18'
      xmlSpace='preserve'
    >
      <path
        fill={color}
        d='M2.668,49.997c-0.595-1.467-1.074-2.987-1.473-4.546H100c0,1.556-0.157,3.065-0.41,4.546H2.668z'
      />
      <path
        fill={color}
        d='M18.775,4.543C23.989,1.655,29.98,0,36.362,0c6.378,0,12.356,1.659,17.564,4.543H18.775z'
      />
      <path
        fill={color}
        d='M0.316,31.816c0.193-1.55,0.495-3.062,0.879-4.543h91.817c1.254,1.393,2.361,2.911,3.307,4.543H0.316z'
      />
      <path
        fill={color}
        d='M2.668,22.727c0.639-1.575,1.394-3.088,2.237-4.546h62.919c0.091,0.154,0.19,0.307,0.279,0.466
c1.51-0.26,3.043-0.466,4.621-0.466c5.571,0,10.75,1.677,15.062,4.546H2.668z'
      />
      <path
        fill={color}
        d='M8.003,13.635c1.318-1.644,2.785-3.155,4.364-4.546h48.01c1.567,1.386,3.015,2.909,4.324,4.546H8.003z'
      />
      <path
        fill={color}
        d='M99.585,40.905H0.316C0.13,39.414,0,37.904,0,36.364h98.413C98.928,37.824,99.325,39.342,99.585,40.905z'
      />
      <path
        fill={color}
        d='M98.413,54.542c-0.56,1.59-1.261,3.107-2.094,4.546H8.003c-1.147-1.43-2.175-2.952-3.098-4.546H98.413z'
      />
      <path
        fill={color}
        d='M12.367,63.634h80.644c-1.545,1.723-3.29,3.261-5.226,4.546h-69.01 C16.465,66.902,14.333,65.361,12.367,63.634z'
      />
    </svg>
  );
};

DarkClouds.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  alt: PropTypes.string
};
DarkClouds.defaultProps = {
  color: '#666666',
  size: '50px',
  alt: 'Dark clouds'
};
export default DarkClouds;
