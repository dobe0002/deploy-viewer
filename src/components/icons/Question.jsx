// From: https://thenounproject.com/search/?q=question&i=2177285

import React from 'react';
import PropTypes from 'prop-types';

const Question = ({ size, color, alt }) => {
  return (
    <svg
      height={size}
      width={size}
      fill={color}
      aria-label={alt}
      xmlns='http://www.w3.org/2000/svg'
      version='1.1'
      x='0px'
      y='0px'
      viewBox='0 0 100 100'
      style={{ enableBackground: 'new 0 0 100 100' }}
      xmlSpace='preserve'
    >
      <g>
        <path d='M42.5,80.5c0,4.7,3.8,8.5,8.5,8.5s8.5-3.8,8.5-8.5S55.7,72,51,72S42.5,75.8,42.5,80.5z M55.5,80.5c0,2.5-2,4.5-4.5,4.5   s-4.5-2-4.5-4.5s2-4.5,4.5-4.5S55.5,78,55.5,80.5z' />
        <path d='M74.9,32.4c-0.7-12-10.4-21.6-22.4-22.4c-6.7-0.4-13.1,1.9-17.9,6.4c-4.8,4.5-7.5,10.9-7.5,17.4c0,1.1,0.9,2,2,2s2-0.9,2-2   c0-5.6,2.2-10.7,6.3-14.5c4.1-3.8,9.4-5.7,14.9-5.4c10,0.6,18,8.6,18.6,18.6c0.5,8.7-4.5,16.6-12.5,19.8C52.7,54.7,49,60,49,66   c0,1.1,0.9,2,2,2s2-0.9,2-2c0-4.3,2.7-8.2,6.8-9.8C69.5,52.4,75.5,42.8,74.9,32.4z' />
      </g>
    </svg>
  );
};

Question.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  alt: PropTypes.string
};
Question.defaultProps = {
  color: '#cccccc',
  size: '50px',
  alt: 'Rain'
};

export default Question;
