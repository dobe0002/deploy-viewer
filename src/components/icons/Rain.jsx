// From: https://thenounproject.com/adamwhitcroft/uploads/?i=2621

import React from 'react';
import PropTypes from 'prop-types';

const Rain = ({ size, color, alt }) => {
  return (
    <svg
      height={size}
      width={size}
      fill={color}
      aria-label={alt}
      xmlns='http://www.w3.org/2000/svg'
      x='0px'
      y='0px'
      viewBox='0 0 100 100'
      enableBackground='new 0 0 99.998 100'
      xmlSpace='preserve'
    >
      <path
        fill={color}
        d='M81.818,71.14v-9.957c5.43-3.146,9.092-9.003,9.092-15.729c0-10.041-8.142-18.182-18.184-18.182
      c-3.638,0-7.005,1.091-9.851,2.934C60.079,18.121,49.299,9.09,36.363,9.09C21.302,9.09,9.092,21.3,9.092,36.361
      c0,8.067,3.526,15.292,9.092,20.285v11.174C7.326,61.528,0,49.813,0,36.361C0,16.279,16.281,0,36.363,0
      C50.01,0,61.886,7.526,68.105,18.646c1.509-0.257,3.041-0.466,4.621-0.466c15.062,0,27.271,12.211,27.271,27.274
      C99.998,57.324,92.406,67.394,81.818,71.14z'
      />
      <path
        fill={color}
        d='M31.819,45.454c2.51,0,4.544,2.038,4.544,4.546v36.363c0,2.513-2.034,4.546-4.544,4.546
      c-2.509,0-4.546-2.033-4.546-4.546V50C27.273,47.491,29.31,45.454,31.819,45.454z'
      />
      <path
        fill={color}
        d='M50,54.546c2.509,0,4.547,2.035,4.547,4.546v36.362C54.548,97.965,52.51,100,50,100
      s-4.546-2.035-4.546-4.546V59.092C45.455,56.579,47.491,54.546,50,54.546z'
      />
      <path
        fill={color}
        d='M68.181,45.454c2.511,0,4.546,2.038,4.546,4.546v36.363c0,2.513-2.035,4.546-4.546,4.546
      c-2.508,0-4.546-2.033-4.546-4.546V50C63.635,47.491,65.673,45.454,68.181,45.454z'
      />
    </svg>
  );
};

Rain.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  alt: PropTypes.string
};
Rain.defaultProps = {
  color: '#ff6d01',
  size: '50px',
  alt: 'Rain'
};

export default Rain;
