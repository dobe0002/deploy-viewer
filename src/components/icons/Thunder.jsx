// From: https://thenounproject.com/adamwhitcroft/uploads/?i=2614

import React from 'react';
import PropTypes from 'prop-types';

const Thunder = ({ size, color, alt }) => {
  return (
    <svg
      height={size}
      width={size}
      fill={color}
      aria-label={alt}
      xmlns='http://www.w3.org/2000/svg'
      x='0px'
      y='0px'
      viewBox='0 0 99.999 100'
      enableBackground='new 0 0 99.999 100'
      xmlSpace='preserve'
    >
      <path
        fill={color}
        d='M72.724,72.729c-0.637,0-1.477,0-2.413,0l8.146-10.028c7.231-2.402,12.45-9.212,12.45-17.248
c0-10.041-8.14-18.178-18.182-18.178c-3.64,0-7.008,1.091-9.849,2.932C60.078,18.125,49.298,9.092,36.363,9.092
c-15.061,0-27.271,12.21-27.271,27.275c0,12.378,8.256,22.817,19.557,26.147L25.582,71.1C10.762,66.505,0,52.694,0,36.365
C0,16.284,16.281,0,36.363,0c13.646,0,25.522,7.528,31.742,18.648c1.509-0.26,3.041-0.466,4.621-0.466
c15.062,0,27.273,12.208,27.273,27.272C99.999,60.517,87.786,72.729,72.724,72.729z'
      />
      <polygon
        fill={color}
        points='45.453,40.908 68.178,40.908 54.542,63.638 70.45,63.638 40.907,100 49.089,72.729 34.088,72.729'
      />
    </svg>
  );
};

Thunder.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  alt: PropTypes.string
};
Thunder.defaultProps = {
  color: '#cc0000',
  size: '50px',
  alt: 'Thunder'
};

export default Thunder;
