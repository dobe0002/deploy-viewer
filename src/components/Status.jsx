import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import map from 'lodash/map';
import Sun from './icons/Sun';
import Rain from './icons/Rain';
import Thunder from './icons/Thunder';
import DarkClouds from './icons/DarkClouds';
import Question from './icons/Question';
import Popover from './Popover';

export default class Status extends React.Component {
  static color(code) {
    switch (code) {
      case 'G':
        return '#38761d';
      case 'O':
        return '#666';
      case 'B':
        return '#ff6d01';
      case 'C':
        return '#cc0000';
      case 'M':
        return '#3f1898';
      default:
        return '#999';
    }
  }

  static icon(code) {
    switch (code) {
      case 'G':
        return <Sun color={this.color(code)} />;
      case 'O':
        return <DarkClouds color={this.color(code)} />;

      case 'B':
        return <Rain color={this.color(code)} />;

      case 'C':
        return <Thunder color={this.color(code)} />;

      default:
        return <Question color={this.color(code)} />;
    }
  }

  static popOverBodyText(appTitle, dateString, note, status = '', code = '') {
    const Label = styled.span`
      display: inline-block;
      padding-right: 10px;
      font-weight: bold;
      &:after {
        content: ':';
      }
    `;
    const UL = styled.ul`
      list-style: none;
      padding-left: 0;
    `;

    return (
      <div>
        {status !== '' && (
          <p>
            {Status.icon(code)}
            <span> </span>
            {status}
          </p>
        )}
        <UL>
          <li>
            <Label>App</Label>
            {appTitle}
          </li>
          <li>
            <Label>Date</Label>
            {dateString}
          </li>
        </UL>
        {note !== '' && (
          <div>
            <hr />
            {note}
          </div>
        )}
      </div>
    );
  }

  popOverTitle(code) {
    const { description, statuses } = this.props;
    if (statuses.length === 0) {
      return (
        <div>
          {Status.icon(code)}
          <span style={{ width: '10px', display: 'inline-block' }} />
          {description}
        </div>
      );
    }

    return <div>Multiple deployments</div>;
  }

  popOverBody() {
    const { appTitle, dateString, note, statuses } = this.props;
    if (statuses.length === 0) {
      return Status.popOverBodyText(appTitle, dateString, note);
    }

    return map(statuses, (status, i) => {
      return (
        <div>
          {i !== 0 && (
            <div
              style={{
                borderTop: '2px solid #000',
                marginTop: '15px',
                marginBottom: '15px'
              }}
            />
          )}

          {Status.popOverBodyText(
            appTitle,
            status.dateString,
            status.note,
            status.statusDescription,
            status.status
          )}
        </div>
      );
    });
  }

  text(code) {
    const { tableCell, popOver, id } = this.props;
    const textStyle = {
      backgroundColor: Status.color(code),
      color: '#fff'
    };
    if (code === 'B') {
      textStyle.color = '#000';
    }
    if (tableCell) {
      if (popOver) {
        textStyle.cursor = 'pointer';
        return (
          // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
          <td style={textStyle} tabIndex='0' id={id}>
            {code}
            <Popover
              id={id}
              popOverTitle={this.popOverTitle(code)}
              body={this.popOverBody()}
            />
          </td>
        );
      }
      return <td style={textStyle}>{code}</td>;
    }
    return (
      <span style={textStyle} className='codeSpan'>
        {code}
      </span>
    );
  }

  render() {
    const { icon, popOver, id, code, description } = this.props;
    if (icon) {
      if (popOver) {
        return (
          // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
          <span id={id} tabIndex='0' style={{ cursor: 'pointer' }}>
            {Status.icon(code)}
            <Popover
              id={id}
              popOverTitle={description}
              body={this.popOverBody()}
            />
          </span>
        );
      }
      return Status.icon(code);
    }
    return this.text(code);
  }
}

Status.propTypes = {
  id: PropTypes.string.isRequired,
  code: PropTypes.string,
  description: PropTypes.string,
  icon: PropTypes.bool,
  tableCell: PropTypes.bool,
  appTitle: PropTypes.string,
  dateString: PropTypes.string.isRequired,
  note: PropTypes.string,
  popOver: PropTypes.bool,
  statuses: PropTypes.arrayOf
};
Status.defaultProps = {
  icon: false,
  description: '',
  tableCell: false,
  popOver: false,
  appTitle: '',
  note: '',
  code: '',
  statuses: []
};
