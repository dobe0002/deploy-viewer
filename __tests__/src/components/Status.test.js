// https://medium.com/@antonioval/making-array-iteration-easy-when-using-async-await-6315c322583

/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Status from '../../../src/components/Status';
import Sun from '../../../src/components/icons/Sun';
import DarkClouds from '../../../src/components/icons/DarkClouds';
import Rain from '../../../src/components/icons/Rain';
import Thunder from '../../../src/components/icons/Thunder';
import Question from '../../../src/components/icons/Question';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('Status component tests', () => {
  const sampleProps = {
    id: 'myStatus',
    code: 'C',
    description: 'This is my status',
    appTitle: 'My application',
    dateString: '10/05/2019',
    note: 'This is a note',
    icon: false,
    tableCell: false,
    popOver: false
  };
  beforeEach(() => {});

  test('Accessibility check - icon', async () => {
    const div = document.createElement('div');
    div.id = 'icon';
    document.body.appendChild(div);
    const wrapper = mount(
      <Status
        icon
        code={sampleProps.code}
        id='icon'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: div }
    );
    const results = await axe(wrapper.getDOMNode());
    expect(results).toHaveNoViolations();
  });

  test('Accessibility check - icon with popover', async () => {
    const div2 = document.createElement('div');
    div2.id = 'iconPopover2';
    document.body.appendChild(div2);
    const wrapper = mount(
      <Status
        icon
        popOver
        code={sampleProps.code}
        id='iconPopover'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: div2 }
    );
    const results = await axe(wrapper.getDOMNode());
    expect(results).toHaveNoViolations();
  });

  test('Accessibility check - table cell without popover', async () => {
    const tr3 = document.createElement('tr');
    tr3.id = 'tablePopover';
    document.body.appendChild(tr3);
    const wrapper = mount(
      <Status
        tableCell
        code={sampleProps.code}
        id='tablePopover'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: tr3 }
    );
    const results = await axe(wrapper.getDOMNode());
    expect(results).toHaveNoViolations();
  });

  test('Accessibility check - table cell with popover', async () => {
    const tr4 = document.createElement('tr');
    tr4.id = 'tablePopover4';
    document.body.appendChild(tr4);
    const wrapper = mount(
      <Status
        tableCell
        popOver
        code={sampleProps.code}
        id='tablePopover5'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: tr4 }
    );
    const results = await axe(wrapper.getDOMNode());
    expect(results).toHaveNoViolations();
  });
  test('Accessibility check - plain text', async () => {
    const div7 = document.createElement('div');
    div7.id = 'icon7';
    document.body.appendChild(div7);
    const wrapper = mount(
      <Status
        code={sampleProps.code}
        id='icon8'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: div7 }
    );
    const results = await axe(wrapper.getDOMNode());
    expect(results).toHaveNoViolations();
  });

  test('Check icon - G', () => {
    const wrapper = shallow(Status.icon('G'));
    const icon = shallow(<Sun color={Status.color('G')} />);
    expect(wrapper).toEqual(icon);
  });

  test('Check icon - O', async () => {
    const wrapper = await shallow(Status.icon('O'));
    const icon = await shallow(<DarkClouds color={Status.color('O')} />);
    expect(wrapper).toEqual(icon);
  });

  test('Check icon - B', () => {
    const wrapper = shallow(Status.icon('B'));
    const icon = shallow(<Rain color={Status.color('B')} />);
    expect(wrapper).toEqual(icon);
  });
  test('Check icon - C', () => {
    const wrapper = shallow(Status.icon('C'));
    const icon = shallow(<Thunder color={Status.color('C')} />);
    expect(wrapper).toEqual(icon);
  });
  test('Check icon - no status sent', () => {
    const wrapper = shallow(Status.icon(''));
    const icon = shallow(<Question color={Status.color('')} />);
    expect(wrapper).toEqual(icon);
  });

  test('Check color - G', () => {
    expect(Status.color('G')).toEqual('#38761d');
  });
  test('Check color - O', () => {
    expect(Status.color('O')).toEqual('#666');
  });
  test('Check color - B', () => {
    expect(Status.color('B')).toEqual('#ff6d01');
  });
  test('Check color - C', () => {
    expect(Status.color('C')).toEqual('#cc0000');
  });
  test('Check color - ""', () => {
    expect(Status.color('')).toEqual('#999');
  });

  test('Snapshot test icon', () => {
    const div = document.createElement('div');
    div.id = 'icon';
    document.body.appendChild(div);
    const wrapper = mount(
      <Status
        icon
        code={sampleProps.code}
        id='icon'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: div }
    );
    expect(wrapper.find('Popover')).toHaveLength(0);
    expect(wrapper.find('td')).toHaveLength(0);
    expect(wrapper.find('Thunder')).toHaveLength(1);
    expect(wrapper.find('.codeSpan')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });

  test('Snapshot test icon with popover', () => {
    const div2 = document.createElement('div');
    div2.id = 'iconPopover2';
    document.body.appendChild(div2);
    const wrapper = mount(
      <Status
        icon
        popOver
        code={sampleProps.code}
        id='iconPopover'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: div2 }
    );
    expect(wrapper.find('Popover')).toHaveLength(1);
    expect(wrapper.find('td')).toHaveLength(0);
    expect(wrapper.find('Thunder')).toHaveLength(1);
    expect(wrapper.find('.codeSpan')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });

  test('Snapshot test cell', () => {
    const tr3 = document.createElement('tr');
    tr3.id = 'tablePopover';
    document.body.appendChild(tr3);
    const wrapper = mount(
      <Status
        tableCell
        code={sampleProps.code}
        id='tablePopover'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: tr3 }
    );
    expect(wrapper.find('Popover')).toHaveLength(0);
    expect(wrapper.find('td')).toHaveLength(1);
    expect(wrapper.find('Thunder')).toHaveLength(0);
    expect(wrapper.find('.codeSpan')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });

  test('Snapshot test cell with popover', () => {
    const tr4 = document.createElement('tr');
    tr4.id = 'tablePopover4';
    document.body.appendChild(tr4);
    const wrapper = mount(
      <Status
        tableCell
        popOver
        code={sampleProps.code}
        id='tablePopover5'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: tr4 }
    );
    expect(wrapper.find('Popover')).toHaveLength(1);
    expect(wrapper.find('td')).toHaveLength(1);
    expect(wrapper.find('Thunder')).toHaveLength(0);
    expect(wrapper.find('.codeSpan')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });

  test('Snapshot test plain text', () => {
    const div7 = document.createElement('div');
    div7.id = 'icon7';
    document.body.appendChild(div7);
    const wrapper = mount(
      <Status
        code={sampleProps.code}
        id='icon8'
        description={sampleProps.description}
        appTitle={sampleProps.appTitle}
        dateString={sampleProps.dateString}
        note={sampleProps.note}
      />,
      { attachTo: div7 }
    );
    expect(wrapper.find('Popover')).toHaveLength(0);
    expect(wrapper.find('td')).toHaveLength(0);
    expect(wrapper.find('Thunder')).toHaveLength(0);
    expect(wrapper.find('.codeSpan')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
