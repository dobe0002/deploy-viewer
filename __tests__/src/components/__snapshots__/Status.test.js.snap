// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component tests Snapshot test cell 1`] = `
<Status
  appTitle="My application"
  code="C"
  dateString="10/05/2019"
  description="This is my status"
  icon={false}
  id="tablePopover"
  note="This is a note"
  popOver={false}
  statuses={Array []}
  tableCell={true}
>
  <td
    style={
      Object {
        "backgroundColor": "#cc0000",
        "color": "#fff",
      }
    }
  >
    C
  </td>
</Status>
`;

exports[`Status component tests Snapshot test cell with popover 1`] = `
<Status
  appTitle="My application"
  code="C"
  dateString="10/05/2019"
  description="This is my status"
  icon={false}
  id="tablePopover5"
  note="This is a note"
  popOver={true}
  statuses={Array []}
  tableCell={true}
>
  <td
    id="tablePopover5"
    style={
      Object {
        "backgroundColor": "#cc0000",
        "color": "#fff",
        "cursor": "pointer",
      }
    }
    tabIndex="0"
  >
    C
    <BS4Popover
      body={
        <div>
          <ForwardRef(styled.ul)>
            <li>
              <ForwardRef(styled.span)>
                App
              </ForwardRef(styled.span)>
              My application
            </li>
            <li>
              <ForwardRef(styled.span)>
                Date
              </ForwardRef(styled.span)>
              10/05/2019
            </li>
          </ForwardRef(styled.ul)>
          <div>
            <hr />
            This is a note
          </div>
        </div>
      }
      hover={true}
      id="tablePopover5"
      popOverTitle={
        <div>
          <Thunder
            alt="Thunder"
            color="#cc0000"
            size="50px"
          />
          <span
            style={
              Object {
                "display": "inline-block",
                "width": "10px",
              }
            }
          />
          This is my status
        </div>
      }
    >
      <div>
        <Popover
          isOpen={false}
          placement="bottom"
          placementPrefix="bs-popover"
          target="tablePopover5"
          toggle={[Function]}
          trigger="hover focus"
        >
          <TooltipPopoverWrapper
            autohide={false}
            delay={
              Object {
                "hide": 0,
                "show": 0,
              }
            }
            fade={true}
            hideArrow={false}
            innerClassName="popover-inner"
            isOpen={false}
            placement="bottom"
            placementPrefix="bs-popover"
            popperClassName="popover show"
            target="tablePopover5"
            toggle={[Function]}
            trigger="hover focus"
          />
        </Popover>
      </div>
    </BS4Popover>
  </td>
</Status>
`;

exports[`Status component tests Snapshot test icon 1`] = `
<Status
  appTitle="My application"
  code="C"
  dateString="10/05/2019"
  description="This is my status"
  icon={true}
  id="icon"
  note="This is a note"
  popOver={false}
  statuses={Array []}
  tableCell={false}
>
  <Thunder
    alt="Thunder"
    color="#cc0000"
    size="50px"
  >
    <svg
      aria-label="Thunder"
      enableBackground="new 0 0 99.999 100"
      fill="#cc0000"
      height="50px"
      viewBox="0 0 99.999 100"
      width="50px"
      x="0px"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M72.724,72.729c-0.637,0-1.477,0-2.413,0l8.146-10.028c7.231-2.402,12.45-9.212,12.45-17.248
c0-10.041-8.14-18.178-18.182-18.178c-3.64,0-7.008,1.091-9.849,2.932C60.078,18.125,49.298,9.092,36.363,9.092
c-15.061,0-27.271,12.21-27.271,27.275c0,12.378,8.256,22.817,19.557,26.147L25.582,71.1C10.762,66.505,0,52.694,0,36.365
C0,16.284,16.281,0,36.363,0c13.646,0,25.522,7.528,31.742,18.648c1.509-0.26,3.041-0.466,4.621-0.466
c15.062,0,27.273,12.208,27.273,27.272C99.999,60.517,87.786,72.729,72.724,72.729z"
        fill="#cc0000"
      />
      <polygon
        fill="#cc0000"
        points="45.453,40.908 68.178,40.908 54.542,63.638 70.45,63.638 40.907,100 49.089,72.729 34.088,72.729"
      />
    </svg>
  </Thunder>
</Status>
`;

exports[`Status component tests Snapshot test icon with popover 1`] = `
<Status
  appTitle="My application"
  code="C"
  dateString="10/05/2019"
  description="This is my status"
  icon={true}
  id="iconPopover"
  note="This is a note"
  popOver={true}
  statuses={Array []}
  tableCell={false}
>
  <span
    id="iconPopover"
    style={
      Object {
        "cursor": "pointer",
      }
    }
    tabIndex="0"
  >
    <Thunder
      alt="Thunder"
      color="#cc0000"
      size="50px"
    >
      <svg
        aria-label="Thunder"
        enableBackground="new 0 0 99.999 100"
        fill="#cc0000"
        height="50px"
        viewBox="0 0 99.999 100"
        width="50px"
        x="0px"
        xmlSpace="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M72.724,72.729c-0.637,0-1.477,0-2.413,0l8.146-10.028c7.231-2.402,12.45-9.212,12.45-17.248
c0-10.041-8.14-18.178-18.182-18.178c-3.64,0-7.008,1.091-9.849,2.932C60.078,18.125,49.298,9.092,36.363,9.092
c-15.061,0-27.271,12.21-27.271,27.275c0,12.378,8.256,22.817,19.557,26.147L25.582,71.1C10.762,66.505,0,52.694,0,36.365
C0,16.284,16.281,0,36.363,0c13.646,0,25.522,7.528,31.742,18.648c1.509-0.26,3.041-0.466,4.621-0.466
c15.062,0,27.273,12.208,27.273,27.272C99.999,60.517,87.786,72.729,72.724,72.729z"
          fill="#cc0000"
        />
        <polygon
          fill="#cc0000"
          points="45.453,40.908 68.178,40.908 54.542,63.638 70.45,63.638 40.907,100 49.089,72.729 34.088,72.729"
        />
      </svg>
    </Thunder>
    <BS4Popover
      body={
        <div>
          <ForwardRef(styled.ul)>
            <li>
              <ForwardRef(styled.span)>
                App
              </ForwardRef(styled.span)>
              My application
            </li>
            <li>
              <ForwardRef(styled.span)>
                Date
              </ForwardRef(styled.span)>
              10/05/2019
            </li>
          </ForwardRef(styled.ul)>
          <div>
            <hr />
            This is a note
          </div>
        </div>
      }
      hover={true}
      id="iconPopover"
      popOverTitle="This is my status"
    >
      <div>
        <Popover
          isOpen={false}
          placement="bottom"
          placementPrefix="bs-popover"
          target="iconPopover"
          toggle={[Function]}
          trigger="hover focus"
        >
          <TooltipPopoverWrapper
            autohide={false}
            delay={
              Object {
                "hide": 0,
                "show": 0,
              }
            }
            fade={true}
            hideArrow={false}
            innerClassName="popover-inner"
            isOpen={false}
            placement="bottom"
            placementPrefix="bs-popover"
            popperClassName="popover show"
            target="iconPopover"
            toggle={[Function]}
            trigger="hover focus"
          />
        </Popover>
      </div>
    </BS4Popover>
  </span>
</Status>
`;

exports[`Status component tests Snapshot test plain text 1`] = `
<Status
  appTitle="My application"
  code="C"
  dateString="10/05/2019"
  description="This is my status"
  icon={false}
  id="icon8"
  note="This is a note"
  popOver={false}
  statuses={Array []}
  tableCell={false}
>
  <span
    className="codeSpan"
    style={
      Object {
        "backgroundColor": "#cc0000",
        "color": "#fff",
      }
    }
  >
    C
  </span>
</Status>
`;
